### PR TITLE
Add basic JSON support

### PR DESCRIPTION
**Pull request by bkuehnert**
Barring special well-known-types mappings, this commit adds full JSON
support. This includes support for the json_name option, and the printer
flags specified by the proto3 spec.

### CODE REVIEW

File: json.lisp [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+             (if indent
+                 (format stream "~&~V,0T\"~A\": " (+ indent 2) (write-to-string k))
+                 (format stream "\"~A\":"  (write-to-string k)))
+             (print-field-to-json v (map-descriptor-val-class map-descriptor)
+                                  (and indent (+ indent 4)) stream camel-case-p numeric-enums-p)))
+    (if indent
+        (format stream "~&~V,0T}" indent)
+        (format stream "}")))
+
+;;; Parse objects that were serialized using the text format
```
Comment (Slids) [inline json.lisp:168]: json?

**Comment by Slids** [comment]
Warning:
CheckLint found errors.
No mention of parameter JSON-NAME in docstring for macro DEFINE-MAP. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/define-proto.lisp:410
No mention of parameter JSON-NAME in docstring for macro DEFINE-MAP. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/define-proto.lisp:410
Don't add new copyright statements. (COPYRIGHT)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:1
No mention of parameter STREAM in docstring for function SKIP-JSON-VALUE. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:302
No mention of parameter STREAM in docstring for function SKIP-JSON-VALUE. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:302

**Comment by bkuehnert** [comment]
This PR is also missing tests for 64 bit integer types & boolean types (both currently have failures). Thank you to ABCL for exposing that.

**Comment by bkuehnert** [comment]
> This PR is also missing tests for 64 bit integer types & boolean types (both currently have failures). Thank you to ABCL for exposing that.

This is now fixed

**Comment by Slids** [comment]
Lines should be under 100 columns. (LINE-TOO-LONG)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:133
Lines should not end in whitespace. (WHITESPACE-AT-EOL)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:207
Lines should be under 100 columns. (LINE-TOO-LONG)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:133
Lines should not end in whitespace. (WHITESPACE-AT-EOL)
	//depot/google3/third_party/lisp/cl_protobufs/json.lisp:207

**Pull request merged**